src/collections/messages.coffee

```
@@ -21,7 +21,14 @@ exports.markMessageAsRead = (params, callback) ->
 
 exports.bulkDeleteMessages = (params, callback) ->
   if Array.isArray params
-    params = id: params
+    if params[0]? and typeof params[0] is 'object' and
+    @isItem params[0], 'message'
+      params = id: params.map (item) -> item.id
+    else if @isId params[0]
+      params = id: params
+    else
+      throw new TSArgsError 'teamsnap.bulkDeleteMessages',
+        'Must provide an `array` of message `ids` or `message` objects'
   else if typeof params is 'object' and @isItem params, 'message'
```

Naming it messages is a good idea. About trowing an error, My concern was that an error here will halt JS execution, so you will need to add a validation on the HTML form to be sure that you are actually passing ids or messages to delete. I can imagine an implementation where this will be called and no messages are being passed (probably by badly written code, but not sure if its worth throwing an error for that). My thinking was that such a case is better for the end user for it to fail silently. Thoughts?